backbone/src/main/java/org/researchstack/backbone/ResourceManager.java

```diff
@@ -113,4 +122,53 @@ public static ResourceManager getInstance() {
      * Onboarding sections
      */
     public abstract Resource getOnboardingManager();
+
+    /**
+     * The NameJsonProvider is a useful class that utilizes the
+     * ResourceManager to convert simple String names to Json Strings
+     */
+    public static class NameJsonProvider implements ResourceNameToStringConverter {
+
+        private Context context;
+
+        public NameJsonProvider(Context context) {
+            this.context = context;
+        }
+
+        @Override
+        public String getJsonStringForResourceName(String resourceName) {
```

Comment:
Cool. My though on this is the Resource.type thing doesn't help us much. We could access Resources the standard way, providing the full filename + extension, and have helpers such as getResourceAsString getResourceAsBytes, getResourceAsJsonObject, getResourceAsCertificate